### PR TITLE
Jakarta Concurrency 1.1 Specification

### PR DESCRIPTION
**Pull request by smillidge**
Signed-off-by: smillidge <steve.millidge@payara.fish>

##Jakarta Concurrency 1.1 Specification PR
When creating a specification project release review, create two PRs with the content divided as follows.

Include the following in PR#1:
- [x] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of wombat_x.y.pdf
- [x] Specification HTML in the form of wombat_x.y.html
- [x] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [x] For a Release Review, a summary that a Compatible Implementation
      is complete, passes the TCK, and that the TCK includes sufficient
      coverage of the specification. The TCK users guide MUST include
      the instructions to run the compatible implementations used to
      validate the release.  Instructions MAY be by reference.
- [x] The URL of the OSSRH staging repository for the api, javadoc:
https://oss.sonatype.org/content/repositories/staging/jakarta/enterprise/concurrent/jakarta.enterprise.concurrent-api/1.1.2/
- [X] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      http://download.eclipse.org/ee4j/jakartaee-tck/jakartaee8-eftl/promoted/eclipse-concurrency-tck-1.0.0.zip
This staged TCK zip file name will be corrected to eclipse-concurrency-tck-**1.1.0**.zip when it gets promoted. 
- [X] The URL of the compatibility certification request issue:
      https://github.com/eclipse-ee4j/concurrency-api/issues/91

Include the following in PR#2:
- [x] Specification JavaDoc in the wombat/x.y/apidocs directory.

### CODE REVIEW

**Comment by dblevins** [comment]
Spec status is Draft, but should be Final or Final Release. 
Also still references “1.1.2-SNAPSHOT”

File: concurrency/1.1/_index.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<!-- Template for the root page of a specification release -->
+---
+title: "Jakarta Concurrency 1.1"
+date: 2019-07-17T22:03:40+0100
```
Comment (bshannon) [inline concurrency/1.1/_index.md]: Don't need the time component.

File: concurrency/1.1/_index.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+summary: "Jakarta EE 8 Release."
+---
+Jakarta Concurrency provides a specification for using concurrency from application components without compromising container integrity while still preserving the Jakarta EE platform's fundamental benefits.
+
+* [Jakarta Concurrency 1.1 Specification Document](./concurrency_1.1.pdf) (PDF)
+* [Jakarta Concurrency 1.1 Specification Document](./concurrency_1.1.html) (HTML)
+* [Jakarta Concurrency 1.1 Javadoc](./apidocs)
+* [Jakarta Concurrency 1.1 TCK](http://downloads.eclipse.org/jakarta/concurrency/1.1/concurrency_tck_1.1.zip)
```
Comment (bshannon) [inline concurrency/1.1/_index.md]: I believe the zip file name is wrong.

File: concurrency/1.1/_index.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+|Payara                 | &check; |     |          |
+|Red Hat                | &check; |     |          |
+|Tomitribe              | &check; |     |          |
+|EE4J PMC               | &check; |     |          |
+|Participant Members    | &check; |     |          |
+|Committer Members      | &check; |     |          |
+
+# Compatible Implementations
+
+* Eclipse GlassFish 5.1
```
Comment (kwsutter) [inline concurrency/1.1/_index.md]: Should include a link to the CI:  https://www.eclipse.org/downloads/download.php?file=/glassfish/glassfish-5.1.0.zip

**Comment by kwsutter** [comment]
- There is no LICENSE file in the API jar file.
- The javadoc jar file is missing the doc-files/speclicense.html.  When I click on the "license terms" at the bottom of the javadoc pages, this license is missing.

**Comment by smillidge** [comment]
https://github.com/eclipse-ee4j/concurrency-api/pull/92 fixes the license issue in the api jar and javadoc jar once merged I will restage the jars

File: concurrency/1.1/_index.md (hunk 3 above)
Comment (smillidge) [reply concurrency/1.1/_index.md]: Done

**Comment by smillidge** [comment]
@kwsutter javadocs and api jars have now been restaged with correct licenses

File: concurrency/1.1/_index.md (hunk 2 above)
Comment (smillidge) [reply concurrency/1.1/_index.md]: Not sure what it should be as the one currently being used for testing is
http://download.eclipse.org/ee4j/jakartaee-tck/jakartaee8-eftl/promoted/eclipse-concurrency-tck-1.0.0.zip

**Comment by smillidge** [comment]
@bshannon corrected issues on the spec documents I,m not sure what the URL of the final TCK download will be

**Comment by smillidge** [comment]
#10 redone with api javadocs from latest staged jar

**Comment by bshannon** [comment]
> @bshannon corrected issues on the spec documents I,m not sure what the URL of the final TCK download will be

As described in the checklist, the final URL will be:
https://download.eclipse.org/jakartaee/concurrency/1.1/eclipse-concurrency-tck-1.1.0.zip

**Comment by scottkurz** [comment]
Missing `concurrency/_index.md` file.

File: concurrency/1.1/_index.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+<!-- Template for the root page of a specification release -->
```
Comment (scottkurz) [inline concurrency/1.1/_index.md]: Remove this first line to get the table formatting that we see with the other specs

File: concurrency/1.1/_index.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+title: "Jakarta Concurrency 1.1"
+date: 2019-07-17
+summary: "Jakarta EE 8 Release."
+---
+Jakarta Concurrency provides a specification for using concurrency from application components without compromising container integrity while still preserving the Jakarta EE platform's fundamental benefits.
+
+* [Jakarta Concurrency 1.1 Specification Document](./concurrency_1.1.pdf) (PDF)
+* [Jakarta Concurrency 1.1 Specification Document](./concurrency_1.1.html) (HTML)
+* [Jakarta Concurrency 1.1 Javadoc](./apidocs)
+* [Jakarta Concurrency 1.1 TCK](http://downloads.eclipse.org/jakarta/concurrency/1.1/concurrency-tck-1.0.0.zip)
```
Comment (scottkurz) [inline concurrency/1.1/_index.md]: Noticing this doesn't match the link in the PR comment:  `http://download.eclipse.org/ee4j/jakartaee-tck/jakartaee8-eftl/promoted/eclipse-concurrency-tck-1.0.0.zip`   Maybe that's just a staging link and here's what you expect the final URL to be?

File: concurrency/1.1/concurrency_1.1.html [hunk 6]
```diff
@@ -0,0 +1,668 @@
...
+<div id="content">
+<div id="preamble">
+<div class="sectionbody">
+<div class="literalblock">
+<div class="content">
+<pre>Specification: Jakarta Concurrency
+
+Version: 1.1
+
+Status:
```
Comment (scottkurz) [inline concurrency/1.1/concurrency_1.1.html]: Should this "Status" now say "Final Release"?

**Comment by scottkurz** [comment]
Maybe this has been discussed somewhere already but I'd expect the TCK challenge instructions to include a link to the specific repo you file the issue against (presumably https://github.com/eclipse-ee4j/concurrency-api/issues/91).   It just says:  _Test challenges must be made in writing via the Concurrency specification project issue tracker_  which I guess keeps things flexible but expected we'd pin this down.

**Comment by bshannon** [comment]
We decided that was good enough for this release, although in the future an exact URL would be preferred.

**Comment by starksm64** [comment]
For wave 1, this spec should be due by 8/7, so please review this ASAP. Note there is a good checklist for https://github.com/jakartaee/specification-committee/blob/master/spec_review_checklist.md

**Comment by smillidge** [comment]
Not going to happen for wave 1 now I am travelling.

**Comment by starksm64** [comment]
# Spec Review Checklist

1. Spec PR
  - [x] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [x] Directory of form {spec}/x.y
  - [x] PDF of form {spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [x] HTML of form {spec}-spec-x.y.html ("-spec" preferred but not required)
  - [x] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [x] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [x] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [x] Staging repository link of the form https://oss.sonatype.org/conte/repositories/staging/jakarta/{spec}/jakarta-{spec}-api/x.y.z/
  - [x] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [x] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [x] Second PR for just apidocs

2. _index.md
  - [x] Link to spec pdf
  - [x] Link to spec html
  - [x] Link to apidocs
  - [x] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip
  - [x] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [x] Name of and link to at least one Compatible Implementation

3. javadocs
  - [x] Footer contains Eclipse copyright and link to license
  - [x] ESFL license is included, usually as doc-files/speclicense.html
  - [x] no META-INF directory in PR

4. Spec PDF
  - [x] Correct spec title
  - [x] Version number of the form x.y, not x.y.z
  - [x] Correct Eclipse copyright line
  - [x] No DRAFT or SNAPSHOT

5. Spec HTML
  - [x] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [x] EFTL license file, preferably named LICENSE.md
  - [x] User's Guide (or equivalent documentation)
  - [x] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [x] Software requirements listed
  - [x] Installation and configuration described
  - [x] How to run tests
  - [x] Where to file challenges

8. Compatibility certification request
  - [x] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [x] SHA-256 fingerprint matches staged TCK zip file
073de7b8f33895b663ae2dd1025ea1383de595566992c9818547e324f3f378c8
  - [] Request issue has `certification` label.

9. TCK results summary
  - [x] Page is hosted by Compatible Implementation project
  - [x] Includes all information from certification request
  - [x] Summary includes number of tests passed, failed, errors

File: concurrency/1.1/_index.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
+---
+title: "Jakarta Concurrency 1.1"
+date: 2019-07-17
+summary: "Jakarta EE 8 Release."
+---
+Jakarta Concurrency provides a specification for using concurrency from application components without compromising container integrity while still preserving the Jakarta EE platform's fundamental benefits.
+
+* [Jakarta Concurrency 1.1 Specification Document](./concurrency_1.1.pdf) (PDF)
```
Comment (waynebeaton) [inline concurrency/1.1/_index.md]: Link should be concurrency-spec-1.1.pdf

File: concurrency/1.1/_index.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+* [Jakarta Concurrency 1.1 Specification Document](./concurrency_1.1.html) (HTML)
```
Comment (waynebeaton) [inline concurrency/1.1/_index.md]: Link should be concurrency-spec-1.1.html

**Comment by smillidge** [comment]
@kwsutter the GlassFIsh repo is back and therefore so are the TCK results

**Comment by kwsutter** [comment]
> @kwsutter the GlassFIsh repo is back and therefore so are the TCK results.
> I updated the PR description to add a comment RE: the renaming of the promoted TCK

Thanks, @smillidge.  I modified your Description update just slightly to represent what really happened.  :-)  We still need a couple of 1.1.0 TCK update references corrected in the CR and TCK Results.  I added a comment on the CR.  Thanks!

**Comment by kwsutter** [comment]
Okay, the CR was updated and there's a PR in the works to update the TCK Results page.  We're close enough to merge the Spec and Apidocs PRs.

**Comment by smillidge** [comment]
Release PR still needs to be merged in the project as it needs review. Also the Eclipse Jenkins Bot fails IP validation so looks like I will need to recreate the PR. Otherwise all done

**Comment by arjantijms** [comment]
> Also the Eclipse Jenkins Bot fails IP validation so looks like I will need to recreate the PR.

This is a known problem. It should be fixed higher up by the Eclipse web masters, but for now most everyone has used admin rights to override it and merge anyway. The resulting IP log has subsequently been approved later on, so it "should" be fine.

**Comment by bshannon** [comment]
I merged it.

**Pull request merged**